src/core/server/saved_objects/migrations/core/elastic_index.test.ts

```diff
@@ -581,7 +582,12 @@ describe('ElasticIndex', () => {
         })
       );
 
-      const hasMigrations = await Index.migrationsUpToDate(client, index, migrations);
+      const hasMigrations = await Index.migrationsUpToDate(
+        client,
+        index,
+        migrations,
+        kibanaVersion
+      );
       return { hasMigrations };
```

Comment:
Can we add a couple tests where all documents have their `migrationVersion` up to date but  where the new 

```
must_not: {
     term: {
        referencesMigrationVersion: kibanaVersion,
     },
},
```

clause is explicitly true/false? (or would that just not make any sense as we are not querying against a real ES cluster?)

Reply:
> or would that just not make any sense as we are not querying against a real ES cluster?

^ This is my view. We can't tests the result of individual clauses here, all we know is the input and output of the query.